### ISSUE TITLE
metrics collection errors

### ISSUE THREAD

**Issue by bobo333** [spiffe/spire#1107]
* **Version**: 8.1
* **Platform**: docker/kubernetes
* **Subsystem**: agent

The following metrics trigger similar errors in the agent:
- `spire_agent_spire_agent_workload_api_fetch_x509_svid_elapsed_time`
- `spire_agent_workload_api_connection`
- `spire_agent_workload_api_fetch_x509_svid_error`

Ex:
```
time="2019-09-09T23:58:26Z" level=info msg="error gathering metrics: 6 error(s) occurred:\n* collected metric \"spire_agent_spire_agent_workload_api_fetch_x509_svid_elapsed_time\" { label:<name:\"host\" value:\"me.local\" > label:<name:\"registered\" value:\"false\" > label:<name:\"selector\" value:\"k8s:pod-owner-uid:DaemonSet:8e453cd5-d359-11e9-8d42-525400261060\" > label:<name:\"svid_type\" value:\"x509\" > summary:<sample_count:6 sample_sum:0.22155899554491043 > } was collected before with the same name and label values
```

It's mentioned here https://github.com/spiffe/spire/issues/1040, but wanted to make a ticket specific to this error, as that ticket has a broader scope.

**Comment by bobo333** [comment]
this got fixed in `0.8.2`

**Comment by evan2645** [comment]
Thank you @bobo333 I was recently wondering about this :)
